### PR TITLE
Update CHANGELOG in preparation for Irmin 2.0

### PR DESCRIPTION
**Pull request by CraigFe**
This keeps a focus on changes to the API, and ignores quite a few performance improvements / changes under the hood. Perhaps you want to add some references to performance @samoht?

There is also no mention of `irmin-pack` yet.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,48 +1,130 @@
...
 
 **irmin-unix**
 
-- Added `Cli` module to expose some methods to simplify building command-line
-  interfaces using Irmin (517, @zshipko)
-- Switched from custom configuration format to YAML (#504, @zshipko)
-- Added global config file (`~/.irmin/config.yml`)
-- Fixed parsing of commit hashes in `revert` command (#496, @zshipko)
+- Add `Cli` module to expose some methods to simplify building command-line
+  interfaces using Irmin. (517, @zshipko)
```
Comment (avsm) [inline CHANGES.md]:
```suggestion
  interfaces using Irmin. (#517, @zshipko)
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,48 +1,130 @@
...
+
+- Switch from custom configuration format to YAML. (#504, @zshipko)
+
+- Add global config file. (`~/.irmin/config.yml`)
```
Comment (avsm) [inline CHANGES.md]: is there an env var that can override this? worth noting here if so

File: CHANGES.md [hunk 3]
```diff
@@ -1,48 +1,130 @@
...
+
+- Fix parsing of commit hashes in `revert` command. (#496, @zshipko)
 
 **irmin-git**
 
-- Support `ocaml-git` 2.0 (#545, @samoht)
-- Cleanup handling of remote stores (#552, @samoht)
-- Allow import/export of Git repositories using Irmin slices (#561, @samoht)
+- Support `ocaml-git` 2.0. (#545, @samoht)
```
Comment (avsm) [inline CHANGES.md]:
```suggestion
- Require `ocaml-git >= 2.0`. (#545, @samoht)
```

File: CHANGES.md (hunk 2 above)
Comment (CraigFe) [reply CHANGES.md]: From the manpages:

> The irmin.yml is read by default if it is found in the current working directory or defined globally as $HOME/.irmin/config.yml. The configuration file path can also be set using the --config command-line flag.

I'll add some detail here to make this entry more helpful.

**Comment by samoht** [comment]
LGTM

**Pull request merged**